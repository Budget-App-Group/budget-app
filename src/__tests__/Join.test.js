import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import Join from "../components/Join/Join";
import { MemoryRouter } from "react-router";

//contain tests
it("Join does show Sign In button text when mounted", () => {
  const { container } = render(
    <MemoryRouter>
      <Join />
    </MemoryRouter>
  );

  expect(container.textContent).toContain("Sign In");
});

//snapshot tests
describe("Join snapshot testing", () => {
  const component = renderer.create(
    <MemoryRouter>
      <Join />
    </MemoryRouter>
  );
  let tree = component.toJSON();

  test("component Join render matches snapshot", () => {
    expect(tree).toMatchSnapshot();
  });

  test("component Join displays each time rendered", () => {
    expect(tree).toBeTruthy();
  });
});
