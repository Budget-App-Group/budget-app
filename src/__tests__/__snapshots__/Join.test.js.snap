// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Join snapshot testing component Join render matches snapshot 1`] = `
<div
  className="joinOuterContainer"
>
  <div
    className="joinInnerContainer"
  >
    <h1
      className="heading"
    >
      Join
    </h1>
    <div>
      <input
        className="joinInput"
        onChange={[Function]}
        placeholder="name"
        type="text"
      />
    </div>
    <div>
      <input
        className="joinInput mt-20"
        onChange={[Function]}
        placeholder="Room"
        type="text"
      />
    </div>
    <a
      href="/chat?name=&room="
      onClick={[Function]}
    >
      <button
        className="button mt-20"
        type="submit"
      >
        Sign In
      </button>
    </a>
  </div>
</div>
`;
